lib/synapse/haproxy.rb

```diff
@@ -1030,16 +1030,26 @@ def generate_backend_stanza(watcher, config)
       ]
     end
 
+    def haproxy_exec(command)
+      s = UNIXSocket.new(@opts['socket_file_path'])
+      s.write(command)
+      info = s.read
+    ensure
+      if s
+        s.close
+      end
+    end
+
     # tries to set active backends via haproxy's stats socket
     # because we can't add backends via the socket, we might still need to restart haproxy
     def update_backends(watchers)
       # first, get a list of existing servers for various backends
       begin
-        s = UNIXSocket.new(@opts['socket_file_path'])
-        s.write("show stat\n")
-        info = s.read()
+        stat_command = "show stat\n"
+        info = haproxy_exec(stat_command)
```

Comment:
maybe do `info = haproxy_exec(stat_command) || ""` ? that way, if info is something funky, we'll just restart below because we don't find any backends, rather than hitting an error at `info.split`